src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx

```diff
@@ -0,0 +1,368 @@
+    isRequired: true,
+  },
+} satisfies IFieldConfigs;
+
+interface IProductReleaseCreateEditPageProps {
+  isEditPage?: boolean;
+}
+
+export const ProductReleaseCreateEditPage = ({ isEditPage = false }: IProductReleaseCreateEditPageProps) => {
+  const { productReleaseId } = useParams();
+  const { productVersionId } = useParams();
+  const navigate = useNavigate();
+
+  // create page
+  const serviceContainerCreatePage = useServiceContainer(productReleaseApi.createProductRelease);
+
+  // edit page - get method
+  const serviceContainerEditPageGet = useServiceContainer(productReleaseApi.getProductRelease);
+  const serviceContainerEditPageGetRunner = serviceContainerEditPageGet.run;
+
+  // edit page - patch method
+  const serviceContainerEditPagePatch = useServiceContainer(productReleaseApi.patchProductRelease);
+
+  const serviceContainerProductVersion = useServiceContainer(productVersionApi.getProductVersion);
+  const serviceContainerProductVersionRunner = serviceContainerProductVersion.run;
+
+  const { register, setFieldValues, getFieldState, getFieldErrors, handleSubmit, isSubmitDisabled } = useForm();
+
+  const [selectedProductMilestone, setSelectedProductMilestone] = useState<ProductMilestone>();
+
+  const fetchProductMilestones = useCallback(
+    (requestConfig = {}) => {
+      return productVersionApi.getProductMilestones({ id: productVersionId! }, requestConfig);
+    },
+    [productVersionId]
+  );
+
+  const [isSupportLevelSelectOpen, setIsSupportLevelSelectOpen] = useState<boolean>(false);
+
+  useTitle(
+    generatePageTitle({
+      pageType: isEditPage ? 'Edit' : 'Create',
+      serviceContainer: serviceContainerEditPageGet,
+      firstLevelEntity: 'Product',
+      nestedEntity: 'Release',
+      entityName: `${serviceContainerEditPageGet.data?.version} ${PageTitles.delimiterSymbol} <unknown>`,
+    })
+  );
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: {
+          data: {
+            version: serviceContainerProductVersion.data?.version + '.' + data.version,
+            releaseDate: parseDate(data.releaseDate),
+            supportLevel: data.supportLevel,
+            commonPlatformEnumeration: data.commonPlatformEnumeration,
+            productPagesCode: data.productPagesCode,
+            productVersion: { id: serviceContainerProductVersion.data?.id },
+            productMilestone: { id: selectedProductMilestone!.id },
+          },
+        },
+      })
+      .then((response: any) => {
+        const newProductReleaseId = response?.data?.id;
+        if (!newProductReleaseId) {
+          throw new PncError({
+            code: 'NEW_ENTITY_ID_ERROR',
+            message: `Invalid productReleaseId coming from Orch POST response: ${newProductReleaseId}`,
+          });
+        }
+
+        navigate('..');
+      })
+      .catch((error: any) => {
+        console.error('Failed to create Product Release.');
```

Comment:
also log this error to ui logger

Reply:
I might, but we don't do that on other create/edit pages. @matedo1 What do you think?

Reply:
I would keep it as it is. I think it's useless to log it using uiLogger as there is nothing we can do to fix it - we just need to properly inform user that this action was not successful.

Reply:
And the user is informed by the `ServiceContainerCreatingUpdating`.